R/cre.R

```diff
@@ -176,58 +188,75 @@ cre <- function(y, z, X, method_params, hyper_params){
+                                 X_names = X_names,
+                                 include_offset = getElement(method_params,"include_offset"),
+                                 offset_name = getElement(method_params,"offset_name"))
+
+    ite_inf <- ite_list_inf[["ite"]]
+    sd_ite_inf <- ite_list_inf[["sd_ite"]]
+  } else {
+    sd_ite_inf <- NULL
+  }
+
+
+  # Generate rules matrix
   if (length(causal_rules)==0){
     rules_matrix_inf <- NA
-    causal_rules <- NA
+    causal_rules_int <- NA
   } else {
     rules_matrix_inf <- generate_rules_matrix(X_inf, causal_rules)
-    causal_rules <- interpret_select_rules(causal_rules, X_names)
+    causal_rules_int <- interpret_select_rules(causal_rules, X_names)
   }
 
-  # Estimate CATE --------------------------------------------------------------
+  # Estimate CATE
   logger::log_info("Estimating CATE...")
   cate_inf <- estimate_cate(y_inf, z_inf, X_inf, X_names,
-                            getElement(method_params,"include_offset"),
-                            getElement(method_params,"offset_name"),
-                            rules_matrix_inf, causal_rules,
+                            getElement(method_params,"offset"),
+                            rules_matrix_inf, causal_rules_int,
                             getElement(method_params,"cate_method"),
                             ite_inf, sd_ite_inf,
                             getElement(method_params,"cate_SL_library"),
                             getElement(hyper_params,"t_pvalue"))
 
-  M["Causal (significant)"] <- as.integer(length(cate_inf$Rule))-1
+  M["Causal (significant)"] <- as.integer(length(cate_inf$summary$Rule))-1
+
+  # Estimate ITE
+  if (getElement(method_params,"cate_method")=="linreg"){
+    if (!any(is.na(causal_rules_int))){
+      causal_rules_matrix <- generate_rules_matrix(X, causal_rules)
+      causal_rules_matrix <- as.data.frame(causal_rules_matrix) %>%
+        dplyr::transmute_all(as.factor)
+      names(causal_rules_matrix) <- causal_rules_int
+      ite_pred <- predict(cate_inf$model, causal_rules_matrix)
+    } else {
+      ite_pred <- cate_inf$summary$Estimate[1]
+    }
+  } else {
+    # TODO: return predicted ITEs for other CATE Estimators (i.e. DRLearner,...)
+    ite_pred <- NULL
+  }
 
   # Generate final results S3 object
-  results <- list()
-  results[["M"]] <- M
-  results[["CATE"]] <- cate_inf
-  results[["cate_method"]] <- getElement(method_params,"cate_method")
+  results <- list("M" = M,
+                  "CATE" = cate_inf[["summary"]],
+                  "method_params" = method_params,
+                  "hyper_params" = hyper_params,
+                  "ite_pred" = ite_pred)
   attr(results, "class") <- "cre"
 
   # Sensitivity Analysis -------------------------------------------------------
-  #logger::log_info("3 Sensitivity Analysis")
+  #logger::log_info("Starting Sensitivity Analysis")
```

Comment:
In the starting log comments, please use trailing `...`.
`logger::log_info("Starting Sensitivity Analysis ... ")`